### PR TITLE
Fix mistake in OEQThumb and surround with tests and stories

### PR DESCRIPTION
**Pull request by SammyIsConfused**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests and stories have been added

##### Description of change

I made a mistake earlier today on f8edd01, initially intending to include the correct default thumbnail for resource attachments pointing towards links but instead incorrectly defaulting to oeqProvidedThumb which caused generated thumbnails not to be used. 
I wonder if its worth doing a refactor of this component at some point, as the logic is somewhat strange.

As per @edalex-ian 's request, I have surrounded this OEQThumb component with stories and tests.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Review by ChristianMurphy — APPROVED** [review]

**Review by edalex-ian — CHANGES_REQUESTED** [review]
With additional logic it'd be good to have a test for it. But I note this component has no specific tests or stories.

Could you please add a component/OEQThumb story? Feel free to just have enough stories to cover your new setup, plus one or so of the others.

And for bonus points, it'd be pretty useful to have a Jest test around this component - strange it was skipped. :thinking:

**Comment by SammyIsConfused** [comment]
> With additional logic it'd be good to have a test for it. But I note this component has no specific tests or stories.
> 
> Could you please add a component/OEQThumb story? Feel free to just have enough stories to cover your new setup, plus one or so of the others.
> 
> And for bonus points, it'd be pretty useful to have a Jest test around this component - strange it was skipped. thinking

Sounds good, thanks for that. I'll add some tests and stories - this wouldn't have happened if they were there in the first place.

**Comment by SammyIsConfused** [comment]
Tests and stories have been added to surround this component :+1:

File: react-front-end/__stories__/components/OEQThumb.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  htmlAttachment,
+  linkAttachment,
+  resourceLinkAttachment,
+  resourceHtmlAttachment,
+} from "../../__mocks__/OEQThumb.mock";
+import OEQThumb, { OEQThumbProps } from "../../tsrc/components/OEQThumb";
+
+export default {
+  title: "Component/OEQThumb",
+  component: OEQThumb,
```
Comment (PenghaiZhang) [inline react-front-end/__stories__/components/OEQThumb.stories.tsx:34]: Perhaps you can add
```
  argTypes: {
    showPlaceholder: {boolean: false},
  },
```

and then don't need to repeat `showPlaceholder: false,` below.

File: react-front-end/__tests__/tsrc/components/OEQThumb.test.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  htmlAttachment,
+  linkAttachment,
+  resourceHtmlAttachment,
+  resourceLinkAttachment,
+} from "../../../__mocks__/OEQThumb.mock";
+import OEQThumb from "../../../tsrc/components/OEQThumb";
+import * as OEQ from "@openequella/rest-api-client";
+
+describe("<OEQThumb/>", () => {
+  const OEQThumbComponent = (
```
Comment (PenghaiZhang) [inline react-front-end/__tests__/tsrc/components/OEQThumb.test.tsx:33]:
```suggestion
  const buildOEQThumb = (
```

File: react-front-end/__tests__/tsrc/components/OEQThumb.test.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  ])(
+    "%s",
+    (
+      _: string,
+      attachment: OEQ.Search.Attachment,
+      showPlaceHolder: boolean,
+      query: string
+    ) => {
+      const { container } = OEQThumbComponent(attachment, showPlaceHolder);
+      expect(queryForIconId(container, query)).not.toBeNull();
```
Comment (PenghaiZhang) [inline react-front-end/__tests__/tsrc/components/OEQThumb.test.tsx:102]:
```suggestion
      expect(queryForIconId(container, query)).toBeTruthy();
```

File: react-front-end/tsrc/components/OEQThumb.tsx [hunk 4]
```diff
@@ -106,21 +109,34 @@ export default function OEQThumb({
...
-    if (mimeType === "equella/item") {
-      return <Web {...generalThumbStyles} />;
-    }
-    if (mimeType === "equella/link") {
-      return <LinkIcon {...generalThumbStyles} />;
-    }
     let result = defaultThumb;
     if (mimeType?.startsWith("image")) {
-      result = <ImageIcon {...generalThumbStyles} />;
+      result = <ImageIcon id="imageIcon" {...generalThumbStyles} />;
```
Comment (PenghaiZhang) [inline react-front-end/tsrc/components/OEQThumb.tsx:114]: I image there will be multiple image icons that have the this ID in one page?

File: react-front-end/tsrc/components/OEQThumb.tsx [hunk 5]
```diff
@@ -106,21 +109,34 @@ export default function OEQThumb({
...
     } else if (mimeType?.startsWith("video")) {
-      result = <VideoIcon {...generalThumbStyles} />;
+      result = <VideoIcon id="videoIcon" {...generalThumbStyles} />;
```
Comment (PenghaiZhang) [inline react-front-end/tsrc/components/OEQThumb.tsx:116]: Same query as above.

**Review by PenghaiZhang — COMMENTED** [review]
Hi Sam, I have left some queries for you. Thank you!

File: react-front-end/__stories__/components/OEQThumb.stories.tsx (hunk 1 above)
Comment (SammyIsConfused) [reply react-front-end/__stories__/components/OEQThumb.stories.tsx:34]: Done, thanks!

File: react-front-end/tsrc/components/OEQThumb.tsx (hunk 4 above)
Comment (SammyIsConfused) [reply react-front-end/tsrc/components/OEQThumb.tsx]: Yup, good catch. Added the attachment ID to the end, to make it unique.

**Comment by SammyIsConfused** [comment]
Fixed those up @PenghaiZhang , thanks so much :)

**Review by PenghaiZhang — APPROVED** [review]
Thanks Sam. I think it is good to go.

**Review by PenghaiZhang — APPROVED** [review]

**Comment by SammyIsConfused** [comment]
@PenghaiZhang and I looked at and discussed the language string commit in the office and it was all good with him (actually his suggestion). So I'll go ahead and merge this.

**Pull request merged**